pkg/daemon/ceph/osd/remove.go

```diff
@@ -91,7 +91,7 @@ func removeOSD(clusterdContext *clusterd.Context, clusterInfo *client.ClusterInf
 		if err != nil {
 			// If we want to force remove the OSD and there was an error let's break outside of
 			// the loop and proceed with the OSD removal
-			if forceOSDRemoval {
+			if forceRemovalCallback(osdID) {
```

Comment:
We need another return flag from the callback. If the user decides not to continue, we shouldn't continue prompting them every minute if they want to try again. So the downstream call would return `false, true` if the user decides not to force remove it, but by default the upstream will return `false, false` so we will continue trying again. Something like this:
```suggestion
			forceRemoval, exitIfNotSafe := forceRemovalCallback(osdID)
			if forceRemoval {
			  ...
			} else if exitIfNotSafe {
			  // log an error and return
			} else {
			  // sleep...
			}
```

Reply:
Right, we don't want to get ack every minute